File: android/src/main/java/com/supermap/interfaces/utils/SLanguage.java
package com.supermap.interfaces.utils;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.supermap.data.Toolkit;

public class SLanguage extends ReactContextBaseJavaModule {
    public static final String REACT_CLASS = "SLanguage";
    static private String m_language = "CN";
    public SLanguage(ReactApplicationContext context) {
        super(context);
    }
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @ReactMethod
    public void setLanguage(String language,Promise promise) {
        try {
            m_language  = language;
            Toolkit.setLanguage(m_language);
//            Toolkit.setLanguage(language);
            promise.resolve(true);
        } catch (Exception e) {
            promise.reject(e);
        }
    }

    static public String getLanguage(){
        return m_language;
    }
}

File: ios/interfaces/utils/SLanguage.m
//
//  SLanguage.m
//  Supermap
//
//  Created by zhiyan xie on 2019/12/6.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "SLanguage.h"
#import "SuperMap/Toolkit.h"


static NSString* g_language = @"CN";
@implementation SLanguage
RCT_EXPORT_MODULE();

+(NSString*)getLanguage{
    return g_language;
}
RCT_REMAP_METHOD(setLanguage,language:(NSString*)language setLanguageResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
   @try {
       g_language = language;
       [Toolkit setLanguage:g_language];
       NSLog(@"-- current language %@",g_language);
       resolve(@(YES));
   }@catch (NSException *exception) {
       reject(@"SScene", exception.reason, nil);
   }
}

@end
